app/dao/users_dao.py

```diff
@@ -85,56 +93,70 @@ def delete_model_user(user):
     db.session.commit()
 
 
-def delete_user_verify_codes(user):
-    VerifyCode.query.filter_by(user=user).delete()
+def delete_user_verify_codes(user) -> int:
+    stmt = delete(VerifyCode).where(VerifyCode.user == user)
+    rows_deleted = db.session.execute(stmt).rowcount
     db.session.commit()
+    return rows_deleted
 
 
-def count_user_verify_codes(user):
-    query = VerifyCode.query.filter(
+def count_user_verify_codes(user) -> int:
+    stmt = select(func.count()).select_from(VerifyCode).where(
         VerifyCode.user == user,
         VerifyCode.expiry_datetime > datetime.utcnow(),
         VerifyCode.code_used.is_(False)
     )
-    return query.count()
+
+    return db.session.scalars(stmt).first()
 
 
 def verify_within_time(user, age=timedelta(seconds=30)):
-    query = VerifyCode.query.filter(
+    stmt = select(func.count()).select_from(VerifyCode).where(
         VerifyCode.user == user,
         VerifyCode.code_used.is_(False),
         VerifyCode.created_at > (datetime.utcnow() - age)
     )
-    return query.count()
+
+    return db.session.scalars(stmt).first()
 
 
 def get_user_by_id(user_id=None):
-    if user_id:
-        return User.query.filter_by(id=user_id).one()
-    return User.query.filter_by().all()
+    if user_id is None:
+        # Get all users.
+        stmt = select(User)
+        return db.session.scalars(stmt).all()
+
+    # Get one user, or raise an exception.
+    stmt = select(User).where(User.id == user_id)
+    return db.session.scalars(stmt).one()
 
 
 def get_user_by_email(email):
-    return User.query.filter(func.lower(User.email_address) == func.lower(email)).one()
+    stmt = select(User).where(func.lower(User.email_address) == func.lower(email))
```

Comment:
Like "date", I don't see "lower" in the docs.  Rather or not it's valid, should the right side of the comparison be `email.lower()`?

Reply:
I'm sure func.lower magically works the same way func.date works.  I am changed the right side to `email.lower()`.